quadratic/base.py

```diff
@@ -61,13 +61,20 @@ def min_ineq_qp(problem: QuadraticProblem) -> np.ndarray:
         if np.all(p == 0):
             A, _ = combine_linear([eq.c for eq in working_set])
             lambda_vec = np.linalg.solve(A, g)
-            if np.all(lambda_vec >= 0):
+
+            # in the working set we transformed all constraints into equalities, but we want to check for inequalities
+            current_set = [const for const in problem.constraints if const in working_set]
```

Comment:
FYI - Leaky abstraction imo (schon von @tmpr) - wenn man auf Equality überprüft (effektiv durch _const in working_set_),  dann würde ich erwartn, dass nur wenn die 2 Objekte ganz gleich sind die Equality hält. Ein custom __eq__ Override  wo manches ausgelassen wird (in dem fall _equality_type_), verschleiert das. Um zu wissen, dass der Code hier funktioniert, muss man als Entwickler in die __eq__ Methode ansehen. Sauberer wäre eine neue Methode, z.b. LinearConstraint.is_linear_system_equal - das geht dann aber natürlich nicht so schön mit _in working_set_.

Ist nur ne Anmerkung, könnens schon so lassen.

Reply:
Fair, des is eigentlich a schlechte equivalenz :sadge: